user-api/src/main/java/io/enmasse/user/model/v1/UserSpec.java

```diff
@@ -17,7 +17,7 @@
     private final String username;
     private final UserAuthentication authentication;
     private final List<UserAuthorization> authorization;
-    private final Pattern usernamePattern = Pattern.compile("^[a-z0-9]+([a-z0-9_\\-]*[a-z0-9]+|[a-z0-9]*)$");
+    private final Pattern usernamePattern = Pattern.compile("^[a-z0-9]+([a-z0-9_@.\\-]*[a-z0-9]+|[a-z0-9]*)$");
```

Comment:
Are we correct to still disallow uppercase?  What if my organisation uses uppercase usernames?

Reply:
All usernames in keycloak are lowercased, so I'm not sure if we should allow it.

Reply:
I was thinking about the case where the openshift user id (which may or may not be ultimately stored in a keycloak instance), happens to include uppercase.  May be this is not valid concern.

Reply:
@k-wall The federated user fields have no restrictions on the name. This affects only the name of the user we create in keycloak. It can still reference openshift users with any characters in them.

In the case of the realm admin user, the .metadata.name and .spec.username will be lowercased by keycloak-controller, so that it will be accepted by the user api.

Reply:
Thanks for the clarification.